stf.core/src/stf.core/net/adoptopenjdk/stf/extensions/core/StfCoreExtension.java

```diff
@@ -1459,6 +1459,31 @@ public void verifyJavaArgsContains(Stage stage, String... mandatoryArgs) throws
 					+ "Actual Java arguments for this stage are \"" + convertToString(actualArgs) + "\"");
 		}
 	}
+	
+	
+	/**
+	 * Checks if that java processes using the primary JVM contains the given set of JVM options
+	 * 
+	 * @param stage is the stage whose arguments should be checked.
+	 * @param argsToCheck contains 1 or more java arguments.
+	 */
+	public boolean isJavaArgsPresent(Stage stage, String... argsToCheck) throws StfException {
+		if (stage == Stage.INITIALISATION) {
+			throw new StfException("Java argument verification is not valid for the initialisation stage.");
+		}
+
+		// Get hold of the mandatory and actual java arguments
+		List<String> argsToCheckList = Arrays.asList(argsToCheck);
+		ArrayList<String> actualArgs = getActiveJavaArgs(stage);
+
+		// Subtract the actual args from the mandatory. 
+		// The remaining args are mandatories which haven't been given a value.
+		Set<String> argsToCheckSet = new LinkedHashSet<String>(argsToCheckList);
+		Set<String> actualArgsSet = new LinkedHashSet<String>(actualArgs);
+		argsToCheckSet.removeAll(actualArgsSet);
+
+		return !argsToCheckSet.isEmpty();
```

Comment:
Talked with @Mesbah-Alam by call, this !argsToCheckSet.isEmpty() need to update to opposite. And probably need to update String... argsToCheck to String argsToCheck to make it simple and clear.